src/main/java/nl/andrewl/emaildatasetbrowser/view/search/export/QueryExportParameterPanel.java

```diff
@@ -41,66 +43,56 @@ public QueryExportParameterPanel(LuceneSearchPanel searchPanel) {
         });
         add(separateThreadsToggle);
 
-        searchDirectoryButton.addActionListener(e -> {
-            JFileChooser fc = new JFileChooser();
-            fc.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
-            if (fc.showSaveDialog(this) != JFileChooser.APPROVE_OPTION) {
-                return;
-            }
-            this.outputPath = fc.getSelectedFile().toPath();
-            updatePathField();
-        });
-        searchDirectoryButton.setVisible(false);
-        add(searchDirectoryButton);
-
-        searchFileButton.addActionListener(e -> {
-            JFileChooser fc = new JFileChooser();
-            fc.setFileSelectionMode(JFileChooser.FILES_ONLY);
-            fc.setFileFilter(buildFileNameFilter());
-            fc.setAcceptAllFileFilterUsed(true);
-            if (fc.showSaveDialog(this) != JFileChooser.APPROVE_OPTION) {
-                return;
-            }
-            this.outputPath = fc.getSelectedFile().toPath();
-            updatePathField();
-        });
-        add(searchFileButton);
-
-        outputPathField.setEditable(false);
-        add(outputPathField);
-    }
-
-    /**
-     * Updates the displayed path.
-     */
-    private void updatePathField() {
-        if (this.outputPath != null) {
-            this.outputPathField.setText(this.outputPath.toString());
-        } else {
-            this.outputPathField.setText("");
-        }
+        dirSelectField.setVisible(false);
+        add(dirSelectField);
+        add(fileSelectField);
     }
 
     /**
      * Exports the query results.
      */
     public CompletableFuture<Void> export() {
+        String query = this.searchPanel.getQuery();
+        Path outputPath = this.separateThreadsToggle.isSelected()
+                ? this.dirSelectField.getSelectedPath()
+                : this.fileSelectField.getSelectedPath();
+        if (!hasValidParameters(query, outputPath)) {
+            CompletableFuture<Void> cf = new CompletableFuture<>();
+            Exception ex = new IllegalArgumentException(
+                    "Cannot export with variables query: \"%s\" and output path: \"%s\""
+                            .formatted(this.searchPanel.getQuery(), outputPath));
+            cf.completeExceptionally(ex);
+            return cf;
+        }
         QueryExportParams params = new QueryExportParams()
                 .withQuery(this.searchPanel.getQuery())
                 .withMaxResultCount((int) this.maxResultsSpinner.getValue())
                 .withSeparateEmailThreads(this.separateThreadsToggle.isSelected());
         QueryExporter exporter = buildExporter(params);
-        return exporter.export(this.searchPanel.getDataset(), this.outputPath);
+        return exporter.export(this.searchPanel.getDataset(), outputPath);
+    }
+
+    private boolean hasValidParameters(String query, Path outputPath) {
+        // Query validation.
+        if (query == null || query == "") {
```

Comment:
You cannot compare strings with `==` unless you're just comparing their memory locations or hoping that the jvm string cache puts them in the same place. I would suggest doing `if (query == null || query.isBlank())`